### DISCUSSION TITLE
Why some trackers websites are not working

### DISCUSSION THREAD

**Discussion by vj-abishek** [subins2000/p2pt#28] [Q&A]
`WebSocket connection to 'wss://tracker.btorrent.xyz/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED`

` WebSocket connection to 'wss://tracker.novage.com.ua/announce' failed: Error in connection establishment: net::ERR_CONNECTION_TIMED_OUT`

`WebSocket connection to 'wss://wsswt.herokuapp.com/' failed: Error during WebSocket handshake: Unexpected response code: 503`

This is being an issue to connect

Tried to disable adblocker and tried to open in incognito, still no use

**Comment by draeder** [comment]
@vj-abishek there are very few public trackers, and of those, fewer that are reliable. 

You could run your own:

```
    // Create a web sockets signaling server
    let lookup = {}

    const hostname = '0.0.0.0';
    const port = 3001;


    var Server = require('bittorrent-tracker').Server

    var server = new Server({
        udp: false, // enable udp server? [default=true]
        http: true, // enable http server? [default=true]
        ws: true, // enable websocket server? [default=true]
        stats: true, // enable web-based statistics? [default=true]
    })

    server.on('error', function (err) {
        // fatal server error!
        console.log(err.message)
    })

    server.on('warning', function (err) {
        // client sent bad data. probably not a problem, just a buggy client.
        console.log(err.message)
    })

    server.on('listening', function () {
        // fired when all requested servers are listening
        console.log('listening on http port:' + server.http.address().port)
        console.log('listening on ws port:' + server.ws.address().port)
    })

    // start tracker server listening! Use 0 to listen on a random free port.
    server.listen(port, hostname, 'listening')

    // listen for individual tracker messages from peers:
    server.on('start', function (addr) {
        console.log('got start message from ' + addr)
        Object.keys(server.torrents).forEach(hash => {
            lookup[server.torrents[hash].infoHash] = server.torrents[hash].peers.length
            console.log("peers: " + server.torrents[hash].peers.length)
        });
    })

    server.on('complete', function (addr) {})
    server.on('stop', function (addr) {})
```

**Reply by vj-abishek** [reply]
Thanks for answering

**Comment by subins2000** [comment]
Here's a list of public trackers: https://github.com/ngosang/trackerslist/blob/master/trackers_all_ws.txt

Public trackers are ran by different people and organizations. All PeerTube instances run their own trackers, but they restrict access to some particular info hashes (their videos).

Uptime of such public trackers can't be guaranteed. But you can run your own if you have the resources. Read the README.

**Reply by draeder** [reply]
I used similar code to what I posted above to build this - [a Heroku ready BitTorrent tracker server](https://github.com/draeder/p2p-tracker). Heroku has a free tier that is perfect for something like this because you get nearly 1 free month of hours, and 2TB of bandwidth to run the tracker server.
